recipes/mpfr/all/conanfile.py

```diff
@@ -29,148 +34,151 @@ class MpfrConan(ConanFile):
     }
 
     exports_sources = "CMakeLists.txt.in", "patches/**"
-    generators = "cmake"
-
-    _autotools = None
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
 
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
     def config_options(self):
         if self.settings.os == "Windows":
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
 
     def requirements(self):
         if self.options.exact_int == "gmp":
-            self.requires("gmp/6.2.1")
+            self.requires("gmp/6.2.1", transitive_headers=True)
         elif self.options.exact_int == "mpir":
             self.requires("mpir/3.0.0")
```

Comment:
it's strange, why transitive_headers but not transitive_libs? And why not transitive in mpir as well?

Reply:
Hi @SpaceIm,
The GMP headers are needed transitively because the MFPR headers explicitly include them, so MPFR consumers have to be able to see the GMP headers in order to build successfully. The libs are not needed if MPFR is linked statically with GMP. If it is linked dynamically (i.e., where GMP is a shared lib), Conan 2.0 will automatically make them visible to downstream clients.